In [2]:
import pandas as pd

df = pd.read_csv("diabetes.csv")

# Count of each outcome
counts = df["Outcome"].value_counts()

# Percentage of each outcome
percentages = df["Outcome"].value_counts(normalize=True) * 100

print("Counts:")
print(counts)

print("\nPercentages:")
print(percentages)


Counts:
Outcome
0    500
1    268
Name: count, dtype: int64

Percentages:
Outcome
0    65.104167
1    34.895833
Name: proportion, dtype: float64


1. What percentage of patients in this dataset have diabetes?

34.89%

2. Is this dataset balanced or imbalanced? (A balanced dataset has roughly equal classes)

This dataset is moderately imbalanced

3. Why does class imbalance matter for machine learning? (You may need to research this)

Class imbalance matters for machine learning as a model may just achieve high accuracy by predicting the majority class.

4. The Pima Indians have one of the highest rates of diabetes in the world (around 50% in some studies). How does this dataset's diabetes rate compare? What might this tell you about the patient selection criteria?

This dataset’s diabetes rate (~35%) is lower than the reported ~50% prevalence in some Pima populations. This difference likely tells us that the dataset may not include a fully representative sample of the entire Pima population.